pkg/scheduler/core/generic_scheduler.go

```diff
@@ -146,6 +152,24 @@ func (g *genericScheduler) snapshot() error {
 // If it succeeds, it will return the name of the node.
 // If it fails, it will return a FitError error with reasons.
 func (g *genericScheduler) Schedule(ctx context.Context, prof *profile.Profile, state *framework.CycleState, pod *v1.Pod) (result ScheduleResult, err error) {
+	annotation, ok := pod.Annotations["trace.kubernetes.io/context"]
+	if ok {
+		propagator := trace.DefaultHTTPPropagator()
+		reqHeader := &http.Header{}
+		decodedContextBytes, err := base64.StdEncoding.DecodeString(annotation)
+		if err != nil {
+			return result, err
+		}
+		err = json.Unmarshal(decodedContextBytes, reqHeader)
+		if err != nil {
+			return result, err
+		}
+		ctx = propagator.Extract(ctx, reqHeader)
+	}
+	ctx, span := global.TraceProvider().Tracer("new-scheduler").Start(ctx, fmt.Sprintf("genericScheduler: Schedule '%s'", pod.Name))
```

Comment:
curious, why Tracer is called `"new-scheduler"`. Do you want to indicate a version? Would it be better to include k8s version number into the tracer name?

Reply:
This is actually a question I had about opentelemetry, because I couldn't find any way that the tracer name had an effect on tracing. Is there some documentation on how this is used and what standards it should follow?